bootstrap/pkg/client/gcp/gcp.go

```diff
@@ -0,0 +1,562 @@
+	kstypes.DefaultComponents = append(kstypes.DefaultComponents, []string{"cloud-endpoints", "cert-manager", "iap-ingress"}...)
+	kstypes.DefaultParameters["cert-manager"] = []kstypes.NameValue{
+		{
+			Name:  "acmeEmail",
+			Value: email,
+		},
+	}
+	kstypes.DefaultParameters["iap-ingress"] = []kstypes.NameValue{
+		{
+			Name:  "ipName",
+			Value: ipName,
+		},
+		{
+			Name:  "hostname",
+			Value: hostname,
+		},
+	}
+	if kstypes.DefaultParameters["jupyter"] != nil {
+		namevalues := kstypes.DefaultParameters["jupyter"]
+		namevalues = append(namevalues,
+			kstypes.NameValue{
+				Name:  "jupyterHubAuthenticator",
+				Value: "iap",
+			},
+			kstypes.NameValue{
+				Name:  string(kftypes.PLATFORM),
+				Value: gcp.GcpApp.Spec.Platform,
+			},
+		)
+	} else {
+		kstypes.DefaultParameters["jupyter"] = []kstypes.NameValue{
+			{
+				Name:  "jupyterHubAuthenticator",
+				Value: "iap",
+			},
+			{
+				Name:  string(kftypes.PLATFORM),
+				Value: gcp.GcpApp.Spec.Platform,
+			},
+		}
+	}
+	if kstypes.DefaultParameters["ambassador"] != nil {
+		namevalues := kstypes.DefaultParameters["ambassador"]
+		namevalues = append(namevalues,
+			kstypes.NameValue{
+				Name:  string(kftypes.PLATFORM),
+				Value: gcp.GcpApp.Spec.Platform,
+			},
+		)
+	} else {
+		kstypes.DefaultParameters["ambassador"] = []kstypes.NameValue{
+			{
+				Name:  string(kftypes.PLATFORM),
+				Value: gcp.GcpApp.Spec.Platform,
+			},
+		}
+	}
+	if kstypes.DefaultParameters["pipeline"] != nil {
+		namevalues := kstypes.DefaultParameters["pipeline"]
+		namevalues = append(namevalues,
+			kstypes.NameValue{
+				Name:  "mysqlPd",
+				Value: gcp.GcpApp.Name + "-storage-pipeline-db",
+			},
+			kstypes.NameValue{
+				Name:  "nfsPd",
+				Value: gcp.GcpApp.Name + "-storage-pipeline-nfs",
+			},
+		)
+	} else {
+		kstypes.DefaultParameters["pipeline"] = []kstypes.NameValue{
+			{
+				Name:  "mysqlPd",
+				Value: gcp.GcpApp.Name + "-storage-pipeline-db",
+			},
+			{
+				Name:  "nfsPd",
+				Value: gcp.GcpApp.Name + "-storage-pipeline-nfs",
+			},
+		}
+	}
+	kstypes.DefaultParameters["application"] = []kstypes.NameValue{
+		{
+			Name:  "components",
+			Value: "[" + strings.Join(kstypes.QuoteItems(kstypes.DefaultComponents), ",") + "]",
+		},
+	}
+	ks := gcp.Children[kftypes.KSONNET]
+	if ks != nil {
+		ksGenerateErr := ks.Generate(kftypes.ALL, options)
+		if ksGenerateErr != nil {
+			return fmt.Errorf("gcp generate failed for %v: %v", string(kftypes.KSONNET), ksGenerateErr)
+		}
+	} else {
+		return fmt.Errorf("%v not in Children", string(kftypes.KSONNET))
+	}
+	return nil
+}
+
+func (gcp *Gcp) replaceText(regex string, repl string, src []byte) []byte {
+	re := regexp.MustCompile(regex)
+	buf := re.ReplaceAll(src, []byte(repl))
+	return buf
+}
+
+func (gcp *Gcp) generateDMConfigs(options map[string]interface{}) error {
+	appDir := gcp.GcpApp.Spec.AppDir
+	gcpConfigDir := path.Join(appDir, GCP_CONFIG)
```

Comment:
in this place you are trying to create directory as `<appDir>/<GCP_CONFIG`, but we are forced to run `kfctl generate all` in `<appDir>/` already, so you are effectively creating `GCP_CONFIG` in the path of `<appDir>/<appDir>/<GCP_CONFIG>`.  This doesn't seem to be the same as `kfctl.sh`, right?